packages/material-ui/src/useMediaQuery/unstable_useMediaQuery.js

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable camelcase */
+
+import React from 'react';
+
+// This variable will be true once the server-side reconciliation is done.
+let didMount = false;
+
+function unstable_useMediaQuery(queryInput, options = {}) {
+  const query = queryInput.replace('@media ', '');
+  const {
+    defaultMatches: defaultMatchesInput = false,
+    noSsr = false,
+    ssrMatchMedia = null,
+  } = options;
+
+  let defaultMatches = defaultMatchesInput;
+
+  if (didMount || noSsr) {
+    defaultMatches = window.matchMedia(query).matches;
+  } else if (ssrMatchMedia) {
+    defaultMatches = ssrMatchMedia(query).matches;
```

Comment:
We need to add documentation that this must be a side-effect free function if you insist on calling `matchMedia` during render.

Reply:
The new implementation started from you coode snippet. I have made some changes to get all the tests pass. What side effect do you have in mind?